src/components/pages/sharedPage/SharedPage.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback, useEffect, useState } from "react";
+import useAsync from "../../../hooks/useAsync";
+import defaultAvatarImage from "../../../images/Avatar.png";
+import { useUser } from "../../../utils/UserContext";
+import { Navigate } from "react-router-dom";
+import FolderSearchBar from "../../folderSearchBar/FolderSearchBar";
+import SharedFolderInfo from "../../folderUserInfo/SharedFolderInfo";
+import * as S from "./SharedPage.style";
+import CardList from "../../cardList/CardList";
+import LoadingSpinner from "../../LoadingSpinner/LoadingSpinner";
+import { getSampleUserFolder, getSampleUserInfo } from "../../../utils/api";
+
+const SharedPage = () => {
+  const [folderInfo, setFolderInfo] = useState(null);
+  const { setUser } = useUser();
+
+  const [userInfoLoading, userInfoError, getUserInfoAsync] = useAsync(getSampleUserInfo);
+  const [userFolderLoading, userFolderError, getUserFolderAsync] = useAsync(getSampleUserFolder);
+
+  const handleUserInfo = useCallback(async () => {
+    const result = await getUserInfoAsync();
+    if (!result) return;
+
+    setUser(result);
+  }, [getUserInfoAsync, setUser]);
+
+  const handleUserFolder = useCallback(async () => {
+    const result = await getUserFolderAsync();
+    if (!result) return;
+    const { folder } = result;
+    setFolderInfo(folder);
+  }, [getUserFolderAsync, setFolderInfo]);
+
+  useEffect(() => {
```

Comment:
```suggestion
  useEffect(() => {
      const fetchUserInfo = async () => {
      const result = await getUserInfo();
      if (result) {
        setUser(result);
      }
    };

    const fetchUserFolder = async () => {
      const result = await getUserFolder();
      if (result) {
        setFolderInfo(result.folder);
      }
    };

    fetchUserInfo();
    fetchUserFolder();
  }, [getUserInfo, getUserFolder, setUser]);
```

useEffect 내에서 실행 되는 handleUserInfo(), handleUserInfo() 함수가 있고 무한 리랜더링 방지를 위해 다시 각 함수를 메모이제이션 하는 것은 유지보수 하기에 어려운 부분으로 보이네요. 왜냐하면 useCallback의 의존성 배열에 다시 useEffect이 의존하기 때문에 구조가 복잡하게 되어 있어서요.  차라리 useEffect 안에 필요한 로직을 넣는 것이 좋아보입니다. 어짜피 이외에 통신 요청을 하는 부분이 없어서요.